geode-protobuf/src/test/java/org/apache/geode/internal/protocol/AuthenticationIntegrationTest.java

```diff
@@ -143,4 +160,54 @@ public void simpleAuthenticationSucceeds() throws Exception {
         regionsResponse.getResponse().getResponseAPICase());
 
   }
+
+  @Test
+  public void ensureAuthenticationModes() {
+    // This assertion is for the following tests, to make sure they test all authentication modes.
+    HandshakeAPI.AuthenticationMode[] modes =
+        {HandshakeAPI.AuthenticationMode.NONE, HandshakeAPI.AuthenticationMode.SIMPLE};
+    assertTrue(Arrays.equals(modes, HandshakeAPI.AuthenticationMode.class.getEnumConstants()));
+  }
+
+  @Test(expected = IOException.class)
+  public void simpleAuthenticationFailsWithNoopModeSet() throws Exception {
+    System.setProperty("geode.protocol-authentication-mode", "NOOP");
+    setUp();
+
+    ProtobufTestUtilities.buildHandshakeRequest(HandshakeAPI.AuthenticationMode.SIMPLE)
+        .writeDelimitedTo(outputStream);
+
+    HandshakeAPI.HandshakeResponse handshakeResponse =
+        HandshakeAPI.HandshakeResponse.parseDelimitedFrom(inputStream);
+
+    assertFalse(handshakeResponse.getOk());
+    assertNotNull(handshakeResponse.getError());
+
+    // socket is closed, writing should throw an IOException.
+    AuthenticationAPI.SimpleAuthenticationRequest authenticationRequest =
+        AuthenticationAPI.SimpleAuthenticationRequest.newBuilder().setUsername(TEST_USERNAME)
+            .setPassword(TEST_PASSWORD).build();
+    authenticationRequest.writeDelimitedTo(outputStream);
+  }
+
+  @Test(expected = IOException.class)
+  public void noopAuthenticationFailsWithSimpleAuthenticationEnabled() throws Exception {
+    System.setProperty("geode.protocol-authentication-mode", "SIMPLE");
```

Comment:
Do we need system property for authentication?? we should avoid that.

Reply:
Good idea. I think this story requires some thought. We've filed a ticket: 
https://issues.apache.org/jira/browse/GEODE-3729

Reply:
Thanks for taking this up !!!